client/js/core/world.js

```diff
@@ -17,12 +18,10 @@ return function World(scene, conn, clientID, clock) {
 	window.testExposure.chunkManager = chunkManager;
 	window.testExposure.entityManager = entityManager;
 
-	conn.on('debug-ray', processRay);
-
-	function processRay(payload) {
-		var pos = new THREE.Vector3(payload.Pos.X, payload.Pos.Y, payload.Pos.Z);
-		self.addSmallCube(pos);
-	}
+	conn.on(Protocol.MSG_DEBUG_RAY, function(dataView) {
+		var result = Protocol.MsgDebugRay.fromProto(dataView);
```

Comment:
The connection knows what type the message is, doesn't it? Why not just have the marshalling/unmarshalling happen at that layer automatically? (so you get the "result" as an argument to your callback)